### ISSUE TITLE
[a11y]: Some sections are missing in Pagination component when page size is increased

### ISSUE THREAD

**Issue by kaushalnavneet** [carbon-design-system/carbon#12735]
### Package

carbon-components-react

### Operating System

MacOS

### Package version

7.59.5

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The issue is spotted in one of the [AVT issue](https://github.ibm.com/org-ids/roadmap/issues/23744#issuecomment-50782414)s opened against our Pipeline team. It is found that the two sections of the `Pagination` component - `Items per page` and the `Page dropdown selection` are missing when the page size is increased by over 300%:

- Pagination on 100%
![Screen Shot 2022-11-28 at 1 04 45 PM](https://user-images.githubusercontent.com/44249850/204349519-9483c4cd-5ac8-44c8-830b-a6105e83bc74.png)

- Pagination on 300% (and above)
![Screen Shot 2022-11-28 at 1 05 25 PM](https://user-images.githubusercontent.com/44249850/204349500-9539008c-498d-4d83-a56f-d5969d5fca60.png)

The `Pagination` component is 
We import the Pagination component from the [@console/pal](https://github.ibm.com/ibmcloud/pal)/carbon-components-react package. (`"@console/pal": "^1.514.11"`)

### Reproduction/example

Explained in description. (We don't directly use `@carbon/react` or `carbon-components-react` packages in our Pipeline code, we use it through the @console/pal package)

### Steps to reproduce

Increase the browser page size by 300%

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi! This is actually intentional - the non-required parts of the component are hidden when there is not enough horizontal space to properly display them. 

It doesn't look like this has been added to the website yet though - @laurenmrice could you link the docs update issue for this?
